src/lib/input/placeholder.ts

```diff
@@ -0,0 +1,81 @@
+import {
+  Directive,
+  Inject,
+  Input,
+  OpaqueToken,
+  TemplateRef, Component, ViewContainerRef, ComponentRef, HostBinding, ViewChild, HostListener,
+  AfterContentInit, ReflectiveInjector, Injector,
+} from '@angular/core';
+import {PortalHost} from '../core';
+import {ComponentPortal} from '../core/portal/portal';
+import {coerceBooleanProperty} from '../core/coersion/boolean-property';
+
+
+export const MD_PLACEHOLDER_HOST_TOKEN  = new OpaqueToken('mdPlaceholderHost');
```

Comment:
These don't normally have `_TOKEN` at the end (e.g., `NG_VALIDATORS`). 

Add description comment for the token.